eva-server/src/main/java/uk/ac/ebi/eva/server/ws/RegionWSServerV2.java

```diff
@@ -102,21 +114,31 @@ public ResponseEntity getVariantsByRegion(@PathVariable("regionId") String regio
 
         AnnotationMetadata annotationMetadata = getAnnotationMetadataHelper(annotationVepVersion,
                 annotationVepCacheVersion);
-        Integer pageSize = service.countByRegionsAndComplexFilters(regions, filters).intValue();
+        Integer totalNumberOfResults = service.countByRegionsAndComplexFilters(regions, filters).intValue();
 
         List<VariantWithSamplesAndAnnotation> variantEntities;
         List<Resource> resourcesList = new ArrayList<>();
 
-        if (pageSize == 0) {
+        if (totalNumberOfResults == 0) {
             response.setStatus(HttpServletResponse.SC_NOT_FOUND);
-            return new ResponseEntity(new Resources<>(resourcesList), HttpStatus.NOT_FOUND);
+            return new ResponseEntity(new PagedResources<>(resourcesList, new PageMetadata(size, page < 0 ? 0 : page,
+                    totalNumberOfResults)), HttpStatus.NOT_FOUND);
```

Comment:
same as with the genes endpoint, an empty list is more a 204 "no content" than a 404 "not found"

Reply:
Fixed :+1: